packages/arb-token-bridge-ui/src/components/TransferPanel/DepositCardL2Failure.tsx

```diff
@@ -19,11 +28,48 @@ export function DepositCardL2Failure({ tx }: { tx: MergedTransaction }) {
     [isConnectedToArbitrum]
   )
 
+  const getRetryableExpiryDays = useCallback(async () => {
+    let daysTillExpiry: number = 0
+    try {
+      const depositTxReceipt = await l1Signer.provider.getTransactionReceipt(
+        tx.txId
+      )
+      const l1TxReceipt = new L1TransactionReceipt(depositTxReceipt)
+      const l1ToL2Msg = await l1TxReceipt.getL1ToL2Message(l2Signer.provider)
+
+      const now = dayjs()
+      const expiryDate = await l1ToL2Msg.getTimeout()
+
+      daysTillExpiry = dayjs(+expiryDate.toString() * 1000).diff(now, 'days')
+    } catch {
+      daysTillExpiry = 0
+    }
+
+    return daysTillExpiry
+  }, [l1Signer, l2Signer])
```

Comment:
We'll need the same code in the `TransactionsTableDepositRow` component, so I think it makes sense to extract this to `RetryableUtils`. We could have this function signature:

```
async function getRetryableTicketExpirationDate({
  l1TxHash,
  l2Provider,
}): Date
```

Reply:
Makes sense. Done.